src/EntryPayload.py

```diff
@@ -0,0 +1,40 @@
+# It is expected that the payload be a dictionary with the following keys:
+#
+# * userId
+# * trackSlug
+# * timeMs
+# * link
+# * notes
+# * type
+
+'''
+userId, trackSlug, timeMs, link, note, type
+userId = long string gus sent (https://mk64-ad77f-default-rtdb.firebaseio.com/users.json)
+trackSlug = slug from track (https://mk64-ad77f-default-rtdb.firebaseio.com/gamedata/cups.json)
+link = discord link or comment str() = "Auto uploaded"
+note = nopass - created automatically
+type = flap OR 3lap
+'''
+
+
+class EntryPayload():
```

Comment:
I'm not sure why this needs to be a class, instead of a function.

One argument might be that this enables potential future functionality based on the same data, which is fair, I just find it hard to imagine a case where you'd need the data in this class but the resulting dictionary is insufficient.

Reply:
This might be personal preference but I almost always prefer to create a class if there is more than one functionality associated with the data. So here the constructor converts the data to the correct types and then to_dict() exports the data to be sent over the API call.

I prefer doing this then writing a function and having to convert the types in the function call (especially in a dynamically typed language) e.g. to_dict(str(time), trackSlug, 'Auto Updated', etc...) It's just more prone to errors.

Also like you mentioned, it allows for future methods if they come up, e.g. entry_payload.urlEncode() or something like that.

Reply:
Works for me.